### PR TITLE
chore: init database auora module

### PR DESCRIPTION
**Pull request by jsbroks**

### CODE REVIEW

File: modules/database/variables.tf [hunk 1]
```diff
@@ -1,67 +1,81 @@
...
-  default     = 1
+variable "preferred_maintenance_window" {
+  description = "The weekly time range during which system maintenance can occur, in (UTC)"
+  type        = string
+  default     = "sun:05:00-sun:06:00"
 }
 
-variable "db_storage_encrypted" {
-  description = "(Optional) Specifies whether the underlying storage layer should be encrypted"
+variable "storage_encrypted" {
```
Comment (zythosec) [inline modules/database/variables.tf:65]: Database encryption at rest should not be optional. It is a minimum requirement of all security teams/standards/compliance frameworks. We should allow people to bring their own KMS keys, but storage_encrypted should always be true.

File: variables.tf [hunk 2]
```diff
@@ -137,7 +157,7 @@ variable "network_private_subnet_cidrs" {
 variable "kubernetes_public_access" {
   type        = bool
   description = "Indicates whether or not the Amazon EKS public API server endpoint is enabled."
-  default     = true
+  default     = false
```
Comment (zythosec) [inline variables.tf:160]: Nice!

File: modules/database/main.tf [hunk 3]
```diff
@@ -7,60 +7,67 @@ locals {
...
-  vpc_id      = var.network_id
-
-  tags = {
-    Name = "${var.namespace}-metadata-store"
-  }
-}
-
 # Random string to use as master password
 resource "random_string" "master_password" {
   length  = 20
```
Comment (zythosec) [inline modules/database/main.tf:12]: Can we bump this to 32?

**Review by zythosec — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 2 above)
Comment (zythosec) [reply variables.tf:160]: Need to update README with default value

File: examples/public-dns-external/main.tf [hunk 4]
```diff
@@ -18,9 +18,14 @@ module "wandb_infra" {
   public_access = true
   external_dns  = true
 
+  deletion_protection = false
```
[the remark this answers is not in the document]
Comment (jsbroks) [reply examples/public-dns-external/main.tf:21]: > Public access disabled

This example, is an example of public access

File: modules/app_eks/main.tf [hunk 5]
```diff
@@ -84,7 +84,7 @@ resource "aws_iam_role" "node" {
 }
 
 locals {
-  cluster_version = "1.20"
+  cluster_version = "1.21"
```
[the remark this answers is not in the document]
Comment (jsbroks) [reply modules/app_eks/main.tf:87]: I ment to bump it and tested

**Review by zythosec — APPROVED** [review]

File: modules/database/main.tf [hunk 6]
```diff
@@ -7,60 +7,67 @@ locals {
...
   special = false
 }
 
-resource "aws_db_subnet_group" "default" {
-  name       = "${var.namespace}-db-subnets"
-  subnet_ids = var.network_private_subnets
+resource "aws_db_parameter_group" "default" {
+  name        = "${var.namespace}-aurora-db-57-parameter-group"
+  family      = "aurora-mysql5.7"
+  description = "${var.namespace}-aurora-db-57-parameter-group"
```
Comment (annirudh) [inline modules/database/main.tf:19]: Let's add the following DB parameters:

```
parameter {
  name = "binlog_format"
  value = "ROW"
}

parameter {
  name = "performance_schema"
  value = "on"
}

parameter {
  name = "slow_query_log"
  value = "on"
}

parameter {
  name = "long_query_time"
  value = 4
}

parameter {
  name = "max_prepared_stmt_count"
  value = "1048576"
}

parameter {
  name = "innodb_online_alter_log_max_size"
  value = "268435456"
}

parameter {
  name = "max_execution_time"
  value = "60000"
}
```

File: modules/database/main.tf [hunk 7]
```diff
@@ -7,60 +7,67 @@ locals {
...
-  cluster_identifier   = "${var.namespace}-db"
-  db_subnet_group_name = aws_db_subnet_group.default.name
+module "aurora" {
+  source  = "terraform-aws-modules/rds-aurora/aws"
+  version = "6.1.3"
 
+  name           = var.namespace
   engine         = "aurora-mysql"
-  engine_version = "${local.major_mysql_version}.mysql_aurora.${local.aurora_mysql_version}"
+  engine_version = "5.7.12"
```
Comment (annirudh) [inline modules/database/main.tf:34]: This is a fairly old version. Could we upgrade to at least `5.7.34`?

File: modules/database/output.tf [hunk 8]
```diff
@@ -6,26 +6,29 @@ output "password" {
...
 }
 
 output "connection_string" {
-  value = "${local.master_username}:${local.master_password}@${aws_rds_cluster.default.endpoint}/${local.database_name}"
+  value = "${local.master_username}:${local.master_password}@${module.aurora.cluster_endpoint}/${local.database_name}"
 }
 
 output "connection_string_reader" {
-  value = "${local.master_username}:${local.master_password}@${aws_rds_cluster.default.reader_endpoint}/${local.database_name}"
+  value = "${local.master_username}:${local.master_password}@${module.aurora.cluster_reader_endpoint}/${local.database_name}"
```
Comment (annirudh) [inline modules/database/output.tf:28]: This is cool. Not critical to do now, but eventually we might want to route this into gorilla using the `GORILLA_METADATA_STORE_REPLICA` env var.

**Review by annirudh — COMMENTED** [review]
LGTM!

File: modules/database/main.tf (hunk 7 above)
Comment (annirudh) [reply modules/database/main.tf:34]: Nevermind, looks like we use `5.7.12` for Lyft. Not sure if there's a newer engine version Aurora supports.

**Comment by jsbroks** [comment]
This PR is included in version 1.0.0 :tada:

**Pull request merged**